docs/Contributing/visual-studio.rst

```diff
@@ -0,0 +1,169 @@
+use the :code:`ssh` protocol instead.
+
+Now you want to clone the respository. You can get the appropriate command by going to your forked copy in a
+browser, click the code button and then select the SSH option as shown in this sample screenshot:
+
+.. image:: ../_static/images/github_clone_ssh.png
+    :align: center
+    :height: 250
+    :alt: GitHub Clone SSH
+
+
+Bring up Visual Studio. Select :menuselection:`View --> Terminal` to open a terminal in the IDE (if one is not
+already shown at the bottom. Once you have the proper GitHub path, here is the command to clone the repository:
+
+.. code-block:: rst
+
+    PS C:\Users\[user]\Source\Repos> git clone git@github.com:[username]/freeciv21.git
+
+
+This will clone the forked repository to the :file:`C:\\Users\\[User]\\Source\\Repos\\freeciv21` directory on
+your computer.
+
+The final repository setup item is to link the original Longturn project repository to your local area on
+your computer:
+
+.. code-block:: rst
+
+    Repos> cd freeciv21
+    Repos\freeciv21> git remote add upstream https://github.com/longturn/freeciv21.git
+    Repos\freeciv21> git fetch upstream
+    Repos\freeciv21> git fetch upstream master
+
+
+You will also need to set a couple global configuration settings so :code:`git` knows a bit more about you.
+
+.. code-block:: rst
+
+    freeciv21> git config --global user.email "[email address associated with GitHub]"
+    freeciv21> git config --global user.name "[your first and last name]"
+
+
+Build Visual Studio Solution
+============================
+
+Now let's get Visual Studio setup. Select :menuselection:`Git --> Local Repositories --> Open Local Repository`
+and then select the :file:`freeciv21` folder. Visual Studio will take a minute to parse the source tree.
+
+Select :menuselection:`File --> Open --> Folder`. Select the :file:`freeciv21` directory and then Visual Studio
+will parse the files in there. It can take a few mintues for this to complete. When complete, Visual Studio
+will open a :strong:`CMake Overview Pages` tab.
+
+Microsoft provides documentation on CMake in Visual Studio --
+https://docs.microsoft.com/en-us/cpp/build/cmake-projects-in-visual-studio?view=msvc-170
+
+
+Final Steps and Notes
+=====================
+
+As of this writing, there is not a fully integrated setup for configuring, compiling (building) and debugging
+Freeciv21 inside of the Visual Studio user interface. If you open any file (such as :file:`CMakeLists.txt`)
+and make a change, Visual Studio will prompt to :strong:`Generate` updated C++ Intellisense information. If
+selected, Visual Studio will configure using the ``fullrelease`` preset in :file:`CMakePresets.json` using the
+Ninja generator. Ninja only supports MSVC in Visual Studio. However, the Longturn community does not support
+MSVC due to licensing constraints. Instead the Longturn community supports the open source LLVM Clang-Cl
```

Comment:
This makes it sound like ClangCL is fine from a licensing point of view, but I think it uses the Microsoft C++ runtime by default. The real reason for using Clang is that our code require proprietary extensions that MSVC doesn't implement.